test.py

```diff
@@ -216,6 +216,43 @@ def test_get_fen(self):
         result = sf.get_fen("makruk", fen, [], False, False, True)
         self.assertEqual(result, fen)
 
+        # makruk piece honor counting
+        fen = "8/3k4/8/2K1S1P1/8/8/8/8 w - - 0 50"
+        moves = ["g5g6M"]
```

Comment:
Officially in the protocol the promotion piece type is required to be lowercase, uppercase is just allowed for compatibility with some non-compliant GUIs.